platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/ConsensusRoundHandler.java

```diff
@@ -324,6 +347,7 @@ public void consensusRound(final ConsensusRound consensusRound) {
         if (!addedFirstRoundInFreeze && isRoundInFreezePeriod(consensusRound)) {
             addedFirstRoundInFreeze = true;
             enterFreezePeriod.run();
+            platformStatusComponent.processStatusAction(new FreezePeriodEnteredAction(consensusRound.getRoundNum()));
```

Comment:
Updating the platform status to FREEZING used to be synchronous because `enterFreezePeriod.run()` called into `FreezeManager.freezeStarted()`, which updated the platform status directly. This changes it to be asynchronous, so the event creation logic for the tipset algorithm needs to key off of the freeze manager instead of the status.

@cody-littley